packages/flutter/lib/src/rendering/object.dart

```diff
@@ -87,14 +87,19 @@ class PaintingContext extends ClipContext {
   /// Repaint the given render object.
   ///
   /// The render object must be attached to a [PipelineOwner], must have a
-  /// composited layer, and must be in need of painting. The render object's
-  /// layer, if any, is re-used, along with any layers in the subtree that don't
-  /// need to be repainted.
+  /// composited layer, and must be in need of painting.
+  ///
+  /// The composited layer must be reset by calling [resetCompositedChildLayer]
```

Comment:
can we assert that this has been called?

Reply:
done